server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -261,7 +271,7 @@ void getMultipleReposSnapshotInfo(ActionListener<GetSnapshotsResponse> listener)
                     }
                 })
 
-                .addListener(listener.map(ignored -> buildResponse()));
+                .addListener(listener.map(ignored -> buildResponse()), executor, threadPool.getThreadContext());
```

Comment:
Good question, but no :grin: Prior to this change we were forking off the `SNAPSHOT_META` pool before calling `applyAfterPredicateAndAdd`, because iterating through all the snapshots creating a new filtered list is kinda too expensive to do as a single `O(N)` operation on the `SNAPSHOT_META` pool. With this change we apply the predicate to each individual `SnapshotInfo` which means there's `O(N)` constant-cost tasks instead and that's cheap enough not to bother forking it any more. But that means that we have to fork here instead so that we don't do the expensive `buildResponse` work on the `SNAPSHOT_META` pool after collecting all the `SnapshotInfo` objects.